parlai/messenger/core/message_sender.py

```diff
@@ -0,0 +1,279 @@
+        'Payload length {} greater than the max of {}'.format(
+            len(payload), MAX_POSTBACK_CHARS
+        )
+    )
+    return {'content_type': 'text', 'title': title, 'payload': payload}
+
+
+def create_text_message(text, quick_replies=None):
+    """Return a list of text messages from the given text. If the message is
+    too long it is split into multiple messages.
+    quick_replies should be a list of options made with create_reply_option.
+    """
+    def _message(text_content, replies):
+        payload = {'text': text_content[:MAX_TEXT_CHARS]}
+        if replies:
+            payload['quick_replies'] = replies
+        return payload
+
+    tokens = [s[:MAX_TEXT_CHARS] for s in text.split(' ')]
+    splits = []
+    cutoff = 0
+    curr_length = 0
+    if quick_replies:
+        assert len(quick_replies) <= MAX_QUICK_REPLIES, (
+            'Number of quick replies {} greater than the max of {}'.format(
+                len(quick_replies), MAX_QUICK_REPLIES
+            )
+        )
+    for i in range(len(tokens)):
+        if (curr_length + len(tokens[i]) > MAX_TEXT_CHARS):
+            splits.append(_message(' '.join(tokens[cutoff:i]), None))
+            cutoff = i + 1
+            curr_length = 0
+        curr_length += len(tokens[i]) + 1
+    if cutoff < len(tokens):
+        splits.append(_message(' '.join(tokens[cutoff:]), quick_replies))
+    return splits
+
+
+def create_attachment_message(attachment_item, quick_replies=None):
+    """Create a message list made with only an attachment.
+    quick_replies should be a list of options made with create_reply_option.
+    """
+    payload = {'attachment': attachment_item}
+    if quick_replies:
+        assert len(quick_replies) <= MAX_QUICK_REPLIES, (
+            'Number of quick replies {} greater than the max of {}'.format(
+                len(quick_replies), MAX_QUICK_REPLIES
+            )
+        )
+        payload['quick_replies'] = quick_replies
+    return [payload]
+
+
+def create_list_element(element):
+    assert 'title' in element, 'List elems must have a title'
+    ret_elem = {
+        'title': element['title'],
+        'subtitle': '',
+        'default_action': {
+            'type': 'postback',
+            'title': element['title'],
+            'payload': element['title'],
+        }
+    }
+    if 'subtitle' in element:
+        ret_elem['subtitle'] = element['subtitle']
+    return ret_elem
+
+
+def create_compact_list_message(raw_elems):
+    elements = [create_list_element(elem) for elem in raw_elems]
+    return {
+        'type': 'template',
+        'payload': {
+            'template_type': 'list',
+            'top_element_style': 'COMPACT',
+            'elements': elements,
+        }
+    }
+
+
+class MessageSender():
+    """MesageSender is a wrapper around the facebook messenger requests that
+    simplifies the process of sending content.
+    """
+
+    """MessageSocket is a wrapper around websocket to simplify message sends
```

Comment:
you might have left this docstring here by accident

Reply:
Good catch here!